inginious/frontend/plugins/task_editorial/static/task_editorial_preview.js

```diff
@@ -25,17 +26,44 @@ function setSolutionNotebook(){
         const notebookContent = JSON.parse(result);
         const solutionNotebookContainer = $("#solution_notebook");
         render_notebook(notebookContent, solutionNotebookContainer);
+        sendSeeTaskSolution();
     });
 }
 
+//Send analytics when a user see the task editorial - tutorial just one time by load data in modal
+function sendSeeTaskTutorial(){
+    $.post("/api/analytics/", {
+        service: {
+            key: "see_task_tutorial",
+            name: "Task editorial - Tutorial"
+        },
+        course_id: getCourseId()
+    })
+}
+
+//Send analytics when a user see the task editorial - solution everytime he enter the task for the first time
+function sendSeeTaskSolution(){
+    $.post("/api/analytics/", {
+        service: {
+            key: "see_task_solution",
+            name: "Task editorial - Solution"
+        },
+        course_id: getCourseId()
+    })
+}
+
+$("#task_tutorial_modal").one("shown.bs.modal", function () {
+    sendSeeTaskTutorial();
+});
+
 //Load solution editor/notebook after the solution modal
 
-$("#task_solution_modal").on("shown.bs.modal", function () {
+$("#task_solution_modal").one("shown.bs.modal", function () {
```

Comment:
The content of the notebook solution was loaded through the API request each time the user opened the solution modal, so this change allows that content to be loaded only once time, instead of load it directly in the template.

Reply:
The content of the notebook solution was loaded through the API request each time the user opened the solution modal, so this change allows that content to be loaded only once time, instead of load it directly in the template.